src/trunk/apps/processing/scevent/eventtool.cpp

```diff
@@ -401,6 +454,24 @@ bool EventTool::init() {
 
 // >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
 bool EventTool::run() {
+
+	if ( _sendClearCache ) {
+		SEISCOMP_DEBUG("Sending clear cache request");
+		ClearCacheRequestMessage cc_msg;
+		connection()->send(&cc_msg);
+		SEISCOMP_DEBUG("Waiting for clear cache response message...");
+		int attempts = 50;
```

Comment:
Note that there could be any number of other messages (picks, origins, ...) in the queue before you receive your response. If the response message is number 51 then you would ignore it. Another approach is a timeout or to let it run until `isExitRequested() == true` which is triggered by sigkill or ctrl+c.

Reply:
I like this, I'll update the code